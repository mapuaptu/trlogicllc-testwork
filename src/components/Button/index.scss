@import '@/styles/_variables.scss';

.button {
  display: inline-flex;
  align-items: center;
  border: 1px solid $color-java;
  border-radius: 5px;
  padding: 5px 10px;
  min-height: 36px;
  font-weight: 700;
  color: $color-java;
  transition: $transition;
  cursor: pointer;

  .icon {
    margin-left: 10px;
  }

  &:hover {
    opacity: 0.6;
  }

  &--fill {
    color: $color-white;
    background-color: $color-java;
  }

  &--disabled {
    opacity: 0.6;
    pointer-events: none;
  }
}
