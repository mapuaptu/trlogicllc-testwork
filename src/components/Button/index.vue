<template>
  <div :class="[$style.button, fill && $style['button--fill']]">
    <span>
      <slot />
    </span>

    <Icon
      v-if="icon"
      :icon="icon"
      :fill="fill ? '#fff': '#14b6bd'"
      :class="$style.icon"
    />
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
