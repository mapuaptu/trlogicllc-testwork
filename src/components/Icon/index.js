import {
  mdiPlusOutline,
  mdiDeleteOutline,
  mdiPencilOutline,
  mdiGithub,
} from '@mdi/js';

const icons = {
  'plus-outline': mdiPlusOutline,
  'delete-outline': mdiDeleteOutline,
  'pencil-outline': mdiPencilOutline,
  github: mdiGithub,
};

export default {
  name: 'Icon',
  props: {
    icon: {
      type: String,
      default: '',
    },
    fill: {
      type: String,
      default: '#fff',
    },
    width: {
      type: Number,
      default: 24,
    },
    height: {
      type: Number,
      default: 24,
    },
  },
  computed: {
    iconPath() {
      return icons[this.icon];
    },
  },
};
