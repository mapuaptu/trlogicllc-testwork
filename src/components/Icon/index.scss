@import '@/styles/_variables.scss';

.icon {
  display: flex;
  align-items: center;
  justify-content: center;
}
