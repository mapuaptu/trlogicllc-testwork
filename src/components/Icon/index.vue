<template>
  <div :class="$style.icon">
    <svg
      :width="width"
      :height="height"
      :viewbox="`0 0 ${width} ${height}`"
    >
      <path
        :d="iconPath"
        :fill="fill"
      />
    </svg>
  </div>
</template>

<script src="./index.js"></script>
<style src="./index.scss" lang="scss" module></style>
