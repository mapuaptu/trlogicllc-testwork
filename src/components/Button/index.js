import Icon from '@/components/Icon/index.vue';

export default {
  name: 'Button',
  components: {
    Icon,
  },
  props: {
    icon: {
      type: String,
      default: '',
    },
    fill: {
      type: Boolean,
      default: false,
    },
  },
};
