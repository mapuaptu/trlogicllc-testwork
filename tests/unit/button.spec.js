import Button from '@/components/Button/index.vue';
import Icon from '@/components/Icon/index.vue';
import { mount } from '@vue/test-utils';

describe('Button component', () => {
  it('1. Rendered success', async () => {
    const wrapper = await mount(Button);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('2. Render Icon component inside success', async () => {
    const wrapper = await mount(Button, {
      propsData: {
        icon: 'delete-outline',
      },
    });

    const icon = wrapper.find(Icon);

    expect(icon.exists()).toBe(true);
  });
});
