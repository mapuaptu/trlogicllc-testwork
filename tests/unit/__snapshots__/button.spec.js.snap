// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component 1. Rendered success 1`] = `
<div class=""><span></span>
  <!---->
</div>
`;
